proto/regen/ecocredit/marketplace/v1/tx.proto

```diff
@@ -174,6 +181,15 @@ message MsgBuyDirect {
     //
     // Since Revision 1
     string retirement_reason = 7;
+
+    // max_fee_amount is the maximum amount of buyer side fees being paid to the marketplace.
+    // If the marketplace fees end up being greater than this amount, the transaction will fail.
+    // If this field is left empty, the marketplace fees will be calculated based on the bid price
+    // and quantity and automatically deducted from the buyer's account.
+    // Fees are always paid in the same denomination as the bid price.
+    //
+    // Since Revision 3
+    string max_fee_amount = 8;
```

Comment:
Actually this isn't a `cosmos.Dec` because we a GDA compliant decimal implementation behind the scenes rather than the SDK's hacky fixed decimal. Okay to leave this off for now?

Reply:
Actually, nevermind - it's actually an integer.